auth/providers/azure/pop_tokenverifier_test.go

```diff
@@ -237,65 +240,140 @@ func GeneratePoPToken(ts int64, hostName, kid string) (string, error) {
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "Invalid token. 'typ' claim is missing")
 
-	expiredToken, _ := GeneratePoPToken(time.Now().Add(time.Minute*-20).Unix(), "", "")
+	expiredToken, _ := GeneratePoPToken(time.Now().Add(time.Minute*-20).Unix(), "", "", "")
 	_, err = verifier.ValidatePopToken(expiredToken)
 	assert.NotNilf(t, err, "PoP verification succeed.")
 
-	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "", tsClaimsMissing)
+	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "", tsClaimsMissing, "")
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "Invalid token. 'ts' claim is missing")
 
-	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "wrongHostnme", "")
+	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "wrongHostnme", "", "")
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "Invalid Pop token due to host mismatch. Expected: \"testHostname\", received: \"wrongHostnme\"")
 
-	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "", uClaimsMissing)
+	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "", uClaimsMissing, "")
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "Invalid token. 'u' claim is missing")
 
-	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", cnfClaimsMissing)
+	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", cnfClaimsMissing, "")
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "Invalid token. 'cnf' claim is missing")
 
-	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", cnfJwkClaimsWrong)
+	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", cnfJwkClaimsWrong, "")
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "Invalid token. 'jwk' claim is empty")
 
-	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", atCnfClaimMissing)
+	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", atCnfClaimMissing, "")
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "could not retrieve 'cnf' claim from access token")
 
-	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", atCnfClaimWrong)
+	invalidToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", atCnfClaimWrong, "")
 	_, err = verifier.ValidatePopToken(invalidToken)
 	assert.EqualError(t, err, "PoP token validate failed: 'cnf' claim mismatch")
+
+	verifierNonce := NewPoPVerifier("testHostname", 1*time.Second)
+	verifierNonce.mapCacheRetentionBuffer = 0
+	// Testing map cache cleanup
+	validToken, _ = GeneratePoPToken(time.Now().Unix(), "testHostname", "", "randomnonce1")
+	_, err = verifierNonce.ValidatePopToken(validToken)
+	assert.NoError(t, err)
+	// Wait for cache to be cleaned up
+	time.Sleep(2 * time.Second)
+
+	_, err = verifierNonce.ValidatePopToken(validToken)
+	assert.NoError(t, err)
```

Comment:
Shouldn't this have an error since the PoP token is only valid for 1 second according to the `NewPoPVerifier` call on line 313, and it's been 2 seconds since the token was created?